src/lib.rs

```diff
@@ -759,14 +758,14 @@ mod unix {
 	}
 
 	fn write_termios(f: impl Fn(&mut Termios)) -> Result<(), Error> {
-		if isatty(STDIN_FILENO)? {
-			let mut t = tcgetattr(STDIN_FILENO)?;
+		if isatty(stdin().as_raw_fd())? {
+			let mut t = tcgetattr(stdin().as_fd())?;
 			reset_termios(&mut t);
 			f(&mut t);
-			tcsetattr(STDIN_FILENO, TCSANOW, &t)?;
+			tcsetattr(stdin().as_fd(), TCSANOW, &t)?;
```

Comment:
I'm not certain if this is the correct way to do this, but it was the easiest way I could find that didn't involve unsafe.

Reply:
(These methods changed to requiring `AsFd` rather than `AsRawFd`.)